server/tests/stt_tt_new_ninjs_test.py

```diff
@@ -0,0 +1,155 @@
+
+    def test_item_type_and_version_fields(self):
+        self.assertEqual(self.item["type"], "picture")
+        self.assertIn("guid", self.item)
+        self.assertIn("uri", self.item)
+        self.assertIn("mimetype", self.item)
+
+    def test_external_id_and_description(self):
+        # This should be the picture association's URI
+        self.assertEqual(
+            self.item["uri"],
+            "http://tt.se/media/image/sdlr1tDM7WXjoU-crop_w2264_h1273_x522_y789",
+        )
+
+    def test_associations_removed(self):
+        self.assertNotIn("associations", self.item)
+
+    def test_versioncreated_absent_for_picture_items(self):
+        """Picture items should not set versioncreated; timestamp comes from parent text item."""
+        self.assertEqual(self.item.get("type"), "picture")
+        self.assertNotIn("versioncreated", self.item)
+
+    def test_versioncreated_capping_method_directly(self):
+        """Test the _cap_versioncreated_to_parent method directly."""
+        parser = self.parser_class()
+
+        # Test data with text association from fixture
+        ninjs_data = {
+            "associations": {
+                "text_item": {
+                    "type": "text",
+                    "versioncreated": "2024-10-03T14:32:36+02:00",  # From fixture
+                    "uri": "http://tt.se/media/text/241003-finlanduleaborgkniv-9fc4edfa",
+                }
+            }
+        }
+
+        # Item with later versioncreated
+        from dateutil.parser import isoparse
+
+        item = {"versioncreated": isoparse("2024-10-03T16:00:00+02:00")}
+
+        # Call the capping method
+        parser._cap_versioncreated_to_parent(item, ninjs_data)
+
+        # The versioncreated should be capped to the parent text item's timestamp
+        expected_time = isoparse("2024-10-03T14:32:36+02:00")
+        self.assertEqual(item["versioncreated"], expected_time)
+
+    def test_versioncreated_no_capping_when_no_parent(self):
+        """Test that versioncreated is preserved when no parent text item exists."""
+        # Create a picture item without text associations
+        test_data = {
+            "uri": "http://tt.se/media/image/test-picture",
+            "type": "picture",
+            "versioncreated": "2024-10-03T16:00:00+02:00",
+            "associations": {
+                "other_picture": {
+                    "type": "picture",  # Not a text item
+                    "versioncreated": "2024-10-03T14:32:36+02:00",
+                }
+            },
+        }
+
+        parser = self.parser_class()
+        result = parser._transform_from_ninjs(test_data)
+
+        # The versioncreated should remain unchanged since no parent text item exists
+        from dateutil.parser import isoparse
+
+        expected_time = isoparse("2024-10-03T16:00:00+02:00")
+        self.assertEqual(result["versioncreated"], expected_time)
+
+    def test_versioncreated_capping_with_multiple_parent_candidates(self):
```

Comment:
I don't think a picture can be a parent to multiple text items, the link is usually the other way around.
or is this a real scenario? is there such content?